src/compiler/checker.ts

```diff
@@ -12476,6 +12471,20 @@ namespace ts {
             }
         }
 
+        function getDiscriminationResultForProperty(target: UnionType, discriminatingType: Type, propertyName: __String, match: Type | undefined, related: (source: Type, target: Type) => boolean | Ternary) {

```

Comment:
Please break this line. And it might be nice to give the return type since it has the ooband value "mismatch".